tests/functional/tests/imu_kit/suite_imu_kit.cpp

```diff
@@ -0,0 +1,130 @@
+
+#include "rtos/ThisThread.h"
+
+#include "CoreAccelerometer.h"
+#include "CoreGyroscope.h"
+#include "CoreI2C.h"
+#include "CoreLSM6DSOX.h"
+#include "EventLoopKit.h"
+#include "IMUKit.h"
+#include "tests/config.h"
+
+using namespace leka;
+using namespace std::chrono;
+using namespace boost::ut;
+using namespace boost::ut::bdd;
+
+suite suite_lsm6dsox = [] {
+	constexpr auto default_min_bound_pitch = -5._f;
+	constexpr auto default_max_bound_pitch = 5._f;
+	constexpr auto default_min_bound_roll  = -5._f;
+	constexpr auto default_max_bound_roll  = 5._f;
+	constexpr auto default_min_bound_yaw   = 170._f;
+	constexpr auto default_max_bound_yaw   = 190._f;
+	constexpr auto measures_number		   = 100;
+	constexpr auto maximal_yaw_drift	   = 0.1F;
+
+	auto i2c		= CoreI2C(PinName::SENSOR_IMU_TH_I2C_SDA, PinName::SENSOR_IMU_TH_I2C_SCL);
+	auto lsm6dsox	= CoreLSM6DSOX {i2c};
+	auto accel		= CoreAccelerometer {lsm6dsox};
+	auto gyro		= CoreGyroscope {lsm6dsox};
+	auto event_loop = EventLoopKit {};
+
+	auto imukit = IMUKit {event_loop, accel, gyro};
+
+	lsm6dsox.init();
+
+	imukit.start();
+
+	"Initialization"_test = [&] {
+		expect(neq(&imukit, nullptr));
+		imukit.init();
+	};
+
+	"Start"_test = [&] {};
+
+	"get angles"_test = [&] {
+		imukit.start();
+
+		"pitch between min/max bounds"_test = [&] {
+			auto [pitch, roll, yaw] = imukit.getAngles();
+
+			expect(pitch > default_min_bound_pitch)
+				<< "Pitch (" << pitch << ") lesser than minimal bound (" << default_min_bound_pitch << ")";
+			expect(pitch < default_max_bound_pitch)
+				<< "Pitch (" << pitch << ") greater than maximal bound (" << default_max_bound_pitch << ")";
+			expect(roll > default_min_bound_roll)
+				<< "Roll (" << roll << ") lesser than minimal bound (" << default_min_bound_roll << ")";
+			expect(roll < default_max_bound_roll)
+				<< "Roll (" << roll << ") greater than maximal bound (" << default_max_bound_roll << ")";
+			expect(yaw > default_min_bound_yaw)
+				<< "Yaw (" << yaw << ") lesser than minimal bound (" << default_min_bound_yaw << ")";
+			expect(yaw < default_max_bound_yaw)
+				<< "Yaw (" << yaw << ") greater than maximal bound (" << default_max_bound_yaw << ")";
+		};
+
+		"pitch, roll, yaw values change over time"_test = [&] {
+			auto i_batch = std::vector<float> {};
+			auto f_batch = std::vector<float> {};
+
+			for (auto i = 0; i < 10; ++i) {
+				auto [pitch, roll, yaw] = imukit.getAngles();
+
+				i_batch.push_back(pitch);
+				i_batch.push_back(roll);
+				i_batch.push_back(yaw);
+
+				rtos::ThisThread::sleep_for(100ms);
+			}
+
+			rtos::ThisThread::sleep_for(100ms);
+
+			for (auto i = 0; i < 10; ++i) {
+				auto [pitch, roll, yaw] = imukit.getAngles();
+
+				f_batch.push_back(pitch);
+				f_batch.push_back(roll);
+				f_batch.push_back(yaw);
+
+				rtos::ThisThread::sleep_for(100ms);
+			}
+
+			expect(neq(i_batch, f_batch));
+		};
+	};
+
+	scenario("imu - drift") = [&] {
+		given("angles measurement is reset") = [&] {
```

Comment:
```suggestion
		given("a new origin is set") = [&] {
```

see previous comment